src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -25,9 +25,15 @@ public string Serialize(object sourceObject)
             {
                 formatter.Serialize(outputStream, sourceObject);
 
-                var outputBytes = outputStream.GetBuffer();
+                var buffer = default(ArraySegment<byte>);
 
-                return Convert.ToBase64String(outputBytes, 0, (int)outputStream.Length);
+#if DOTNET5_4

```

Comment:
Can we roll all of this into a `GetBufferSegment` extension method and reuse that in `HttpUtility` as well?